searx/engines/mastodon.py

```diff
@@ -0,0 +1,75 @@
+# SPDX-License-Identifier: AGPL-3.0-or-later
+# lint: pylint
+"""Mastodon is an open source alternative to large social media platforms
+like Twitter/X, Facebook, ...
+
+Since it's federated and self-hostable, there's a large amount of available
+instances, which can be chosen instead by modifying ``base_url``.
+
+We use their official API for searching, but unfortunately, their Search API
+forbids pagination without OAuth.
+
+That's why we use tootfinder.ch for finding posts, which doesn't support searching
+for users, accounts or other types of content on Mastodon however.
+"""
+
+from urllib.parse import urlencode
+from datetime import datetime
+
+about = {
+    "website": 'https://joinmastodon.org/',
+    "wikidata_id": 'Q27986619',
+    "official_api_documentation": 'https://docs.joinmastodon.org/api/',
+    "use_official_api": True,
+    "require_api_key": False,
+    "results": 'JSON',
+}
+categories = ['social media']
+
+base_url = "https://mastodon.social"
+mastodon_type = "accounts"
+
+number_of_results = 10
+
+
+def request(query, params):
+    args = {
+        'q': query,
+        'resolve': 'false',
+        'type': mastodon_type,
+        'limit': number_of_results,
+    }
+    params['url'] = f"{base_url}/api/v2/search?{urlencode(args)}"
+    return params
+
+
+def response(resp):
+    results = []
+
+    json = resp.json()
+
+    for result in json[mastodon_type]:
+        if mastodon_type == "accounts":
```

Comment:
NP: Consider using .lower() for mastodon_types so the settings.yml can be flexible.

Reply:
Most settings are only working when being configured as lowercase, we would need to lower every occurrence of mastodon_type which seems a bit unnecessary to me, especially since the two working types are already preconfigured in "settings.yml"